### DISCUSSION TITLE
I keep getting "The metadata storage is not up to date" even after running "sync-metadata-storage"

### DISCUSSION THREAD

**Discussion by php4fan** [symfony/symfony#44178] [Q&A]
### Symfony version(s) affected

5.4.0

### Description

I have a relatively simple project where I've done little more than install and configure a few bundles and run some "make:xxxx" commands.

I had formerly succesfully run "make:user" and made and run the corresponding migration creating a user entity.

Now I installed  the `symfonycasts/verify-email-bundle` bundle, and I ran `make:registration-form`.

Afer that I got this success message:
```
 Success! 
           

 Next:
 1) In RegistrationController::verifyUserEmail():
    * Customize the last redirectToRoute() after a successful email verification.
    * Make sure you're rendering success flash messages or change the $this->addFlash() line.
 2) Review and customize the form, controller, and templates as needed.
 3) Run "php bin/console make:migration" to generate a migration for the newly added User::isVerified property.
```

### How to reproduce

So I did (1) and (2) and then:
```
php bin/console make:migration
```

and got this error:
```
In MetadataStorageError.php line 13:
                                                                                                           
  The metadata storage is not up to date, please run the sync-metadata-storage command to fix this issue.  
```
I have no idea what that means. For starters, the "please run the ___ command" message should be more specific and give me the exact command to run:

```
> bin/console sync-metadata-storage

                                                                                                                        
 Command "sync-metadata-storage" is not defined.                                                                        
                                                                                                                        

 Do you want to run "doctrine:migrations:sync-metadata-storage" instead?  (yes/no) [no]:
```
So I chose `yes` and got a success message:
```                                                                                                                    
 [OK] Metadata storage synchronized                                                                                     
```

But then again, when I try to run `make:migration` I keep getting the same error as before:

```
 bin/console make:migration

In MetadataStorageError.php line 13:
                                                                                                           
  The metadata storage is not up to date, please run the sync-metadata-storage command to fix this issue.  
```

I bet I did something wrong, but if that was the case I should have got an understandable error message telling me what exactly I did wrong.

**Comment by php4fan** [comment]
For some reason the error has gone away by changing in `.env.local`:
```
DATABASE_URL="mysql://symfonyexample:1234@127.0.0.1:33306/symfonyexample?serverVersion=10.5"
```
to:
```
DATABASE_URL="mysql://symfonyexample:1234@127.0.0.1:33306/symfonyexample?serverVersion=mariadb-10.5.11"
```
(I skeptically tried that after googling for the issue and finding lots of answers saying that fixes it, with no explanation whatsoever of why)

However, I had succesfully created and run other migrations before with the setting as it was.

**Reply by derrabus** [reply]
So, the problem is solved for you now? `serverVersion=mariadb-10.5.11` is correct if your remote server is a MariaDB 10.5.11. By setting `serverVersion=10.5` you basically tell Doctrine that the remote server is a MySQL 10.5. This might cause strage behavior.

**Comment by derrabus** [comment]
I'm moving this to discussions because this does not appear to be something Symfony can solve.

**Reply by php4fan** [reply]
Can you elaborate on that?

**Reply by derrabus** [reply]
The commands you're using here belong to Doctrine packages. You've posted your issue to the Symfony bug tracker. If you think your problem is a bug that should be solved, you need to report it to Doctrine.
